lapce-core/src/editor.rs

```diff
@@ -210,7 +233,7 @@ impl Editor {
                     .map(|(idx, content)| {
                         let region = &selection.regions()[*idx];
                         (
-                            Selection::region(region.start, region.end),
+                            Selection::region(region.max(), region.max()),
```

Comment:
not sure whether this will break something, but I feel like it makes a bit more sense...

edits after should be applied to the end of the selection right?

Reply:
This will be incorrect if the selection isn't a caret.

Reply:
forgot to ask, when will the selection not be a caret?

Reply:
when ```start != end```

Reply:
I can't workout what I'm missing here. 

for edits_after we never really want to replace the text? I don't think.

Its only used for this + matching pairs (i.e. type "(" it automatically closes it) + I can't think of when this would be wrong